### DISCUSSION TITLE
Использование conventional comments в ревью

### DISCUSSION THREAD

**Discussion by KirillSolovyev** [chocofamilyme/choco-ui#30] [Ideas]
## Проблема:

- Не все комментарии при проверке PR понятны и приходится уточнять у ревьюера что он/она имели ввиду, что отнимает дополнительное время
- Для того, чтобы понять основную суть комментария нужно вчитываться в него и бывают случаи когда нужно перечитывать комментарий несколько раз, что также отнимает время

## Решение:

Использовать [conventional comments](https://conventionalcomments.org/) при написании комментариев, что ускорит процесс код ревью:

- При написании комментария нужно явно указывать что требуется изменить/переписать (actionable comment) в этой части кода: уменьшает количество обращений к ревьюерам для уточнения деталей
- Добавлять префикс для каждого комментария, чтобы лишь по нему можно было понять намерения комментария

Кроме решения проблем, описанных выше у conventional comments есть еще плюсы, описанные у них на сайте

> Конечно, conventional comments не решают всех проблем и у этого подхода есть минусы, но мне кажется, что он лучше всего подходит, особенно для Open source проектов, в которых общение между разработчиками, в основном, идет в GitHub через discussions, issues и PR и ответа ждать приходится долго

**suggestion(review):** Предлагаю использовать [conventional comments](https://conventionalcomments.org/) в процессе код ревью и прописать это в `README` или `CONTRIBUTING`

@Lootjs

**Comment by abdraymd** [comment]
Хорошее предложение, согласен. Но чем больше контрибьюторов, тем сложнее поддерживать это. Может выделить основные префиксы и использовать только их?

**Reply by KirillSolovyev** [reply]
В основном используются 4 типа комментариев: `nitpick`, `suggestion`, `question` и `thought`. Остальные типы используются реже. Думаю, что будет не сложно их объяснить новым ревьюерам и использовать.

Давайте я создам issue с добавлением информации по conventional comments в README? @Lootjs @abdraymd

**Reply by Lootjs** [reply]
я только за ))
